rpcs3/util/vm_native.cpp

```diff
@@ -89,6 +89,24 @@ namespace utils
 	DYNAMIC_IMPORT("KernelBase.dll", MapViewOfFile3, PVOID(HANDLE Handle, HANDLE Process, PVOID Base, ULONG64 Off, SIZE_T ViewSize, ULONG AllocType, ULONG Prot, MEM_EXTENDED_PARAMETER*, ULONG));
 #endif
 
+	long get_page_size()
+	{
+		static const long r = []() -> long
+		{
+#ifdef _WIN32
+			SYSTEM_INFO info;
+			::GetSystemInfo(&info);
+			return info.dwPageSize;
+#else
+			return ::sysconf(_SC_PAGESIZE);
+#endif
+		}();
+
+		ensure(r > 0 && r <= 0x10000);
```

Comment:
Wouldn't it make more sense from a performance pov to put the ensure inside the lambda?

Reply:
It's not executing after startup anyway.